test_data/rules/unreachable_from_main.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N unreachable_from_main`
+
+import 'package:meta/meta.dart';
+
+/// see [Comment]
+main() // OK
+{
+  _f5();
+  f1();
+  f3(() {
+    f4(b);
+  });
+  f4(b);
+  usageInTypeBound();
+  usageInFunctionType();
+  usageInDefaultValue();
+  usageInAnnotation();
+  Future<C5>.value(C5()).extensionUsage();
+  accessors();
+}
+
+class Comment {} // LINT
+
+const a = 1; // LINT
```

Comment:
Can you add a test for multiple variables declared in a single declaration? I guess something like at least one of them is reachable, and at least one is not.

Reply:
done